opentracing/src/main/java/opentracing/Span.java

```diff
@@ -65,29 +63,7 @@
   Span setTag(String key, Number value);
 
   /**
-   * {@code message} is a format string and can refer to fields in the payload by path, like so:
-   *
-   * <pre>{@code
-   *
-   * span.info("first transaction is worth ${transactions[0].amount} ${transactions[0].currency}",
-   *     ImmutableMap.of(
-   *       "transactions", asList(
-   *         Transaction.builder().amount(10).currency("USD").build(),
-   *         Transaction.builder().amount(11).currency("USD").build(),
-   *       )
-   *     )
-   * );
-   * }</pre>
-   *
-   * @param message {@link Formatter format string} that can refer to fields in the args payload.
-   * @param args arbitrary payload
-   */
-  // See https://github.com/opentracing/opentracing.github.io/issues/30 about parameterization
-  Span info(String message, Object... args);
-
-  /** Same as {@link #info}, but for warnings. */
-  Span warning(String message, Object... args);
-
-  /** Same as {@link #info}, but for errors. */
-  Span error(String message, Object... args);
+   * @todo
+   **/
+  Span event(String message, /* @Nullable */ Object payload);
```

Comment:
to clarify, the long form method with 3 args is called `log`, not `event`, because it is a catch-all / last resort method. It's semantic signature is
```java
Span log(long timestamp, String event,  /*@Nullable*/ Object payload);
```
both timestamp and event are required. It specifically exists for people who want to provide their own timestamp (as in historical recording).

[Instant](https://docs.oracle.com/javase/8/docs/api/java/time/Instant.html) is the common way of representing timestamps. I think java-7 has a way to backport those via jsr310

I agree with Adrian that using `long` for the timestamp is the simplest way to avoid getting into the mess of Java time interfaces,